sdk/ml/azure-ai-ml/azure/ai/ml/operations/_environment_operations.py

```diff
@@ -373,43 +378,62 @@ def _get_latest_version(self, name: str) -> Environment:
         )
         return Environment._from_rest_object(result)
 
-    # pylint: disable=no-self-use
-    def _prepare_to_copy(
-        self, environment: Environment, name: Optional[str] = None, version: Optional[str] = None
-    ) -> WorkspaceAssetReference:
-
-        """Returns WorkspaceAssetReference
-        to copy a registered environment to registry given the asset id
+    @monitor_with_activity(logger, "Environment.Share", ActivityType.PUBLICAPI)
+    def share(self, name, version, share_with_name, share_with_version, registry_name) -> Environment:
```

Comment:
Can you add a `*` before `share_with_name` to make it a kw only argument. That will give us a chance to update the parameter name later.